workspaces/local-cli/src/commands/generate/oas.ts

```diff
@@ -86,3 +90,11 @@ export async function emit(
     yaml: yamlPath,
   };
 }
+
+function getStream(stream: any) {
+  return new Promise((resolve) => {
+    const chunks: any = [];
+    stream.on('data', (chunk: any) => chunks.push(Buffer.from(chunk)));
+    stream.on('end', () => resolve(Buffer.concat(chunks).toString()));
+  });
+}
```

Comment:
Given the circumstances of `OasProjectionHelper` needing to do its own serdes and only accepting a string, this is fine. However, with the use of `.on`, and no removing of listeners, this will leak memory (a bit). 

While you get get all fancy with EventEmitters, the thing you can do in Node now is use a `Readable` as an `AsyncIterator`, allowing you to write this and have it take care of all of that for you:

```ts
const chunks: any = [];
for await (let chunk of stream) {
  chunks.push(Bufer.from(chunk));
}
return Buffer.concat(chunks).toString()
```

As a bonus, also that also propagates the stream's errors nicely!